src/libostree/ostree-repo.c

```diff
@@ -2598,6 +2598,19 @@ append_one_remote_config (OstreeRepo      *self,
   return add_remotes_from_keyfile (self, remotedata, path, error);
 }
 
+static gboolean
+min_free_space_str_validate (char *min_free_space_str)
+{
+  GRegex *percent_regex = g_regex_new ("^([0-9]|[1-9][0-9])%$" , 0,0, NULL);
+  GRegex *size_regex = g_regex_new ("^[0-9]+(G|M|T)B$", 0,0, NULL);
+
+  if (g_regex_match (percent_regex, min_free_space_str, 0, NULL) ||
+      g_regex_match (size_regex, min_free_space_str, 0, NULL))
```

Comment:
If we're going to use regexps it feels like we might as well use capture groups to get the data out, then `min_free_space_str_parse()` would just need to handle a `(G|M|T)B` suffix.